plugin-jdbc-sqlite/src/main/java/io/kestra/plugin/jdbc/sqlite/Query.java

```diff
@@ -35,25 +42,79 @@
                 "- id: update",
                 "  type: io.kestra.plugin.jdbc.sqlite.Query",
                 "  url: jdbc:sqlite:myfile.db",
+                "  sqliteFile: {{ outputs.get.outputFiles['myfile.sqlite'] }}",
                 "  sql: select concert_id, available, a, b, c, d, play_time, library_record, floatn_test, double_test, real_test, numeric_test, date_type, time_type, timez_type, timestamp_type, timestampz_type, interval_type, pay_by_quarter, schedule, json_type, blob_type from pgsql_types",
                 "  fetch: true",
                 "- id: use-fetched-data",
                 "  type: io.kestra.plugin.jdbc.sqlite.Query",
                 "  url: jdbc:sqlite:myfile.db",
-                "  sql: \"{% for row in outputs.update.rows %} INSERT INTO pl_store_distribute (year_month,store_code, update_date) values ({{row.play_time}}, {{row.concert_id}}, TO_TIMESTAMP('{{row.timestamp_type}}', 'YYYY-MM-DDTHH:MI:SS.US') ); {% endfor %}\""}
+                "  sqliteFile: {{ outputs.get.outputFiles['myfile.sqlite'] }}",
+                "  sql: \"{% for row in outputs.update.rows %} INSERT INTO pl_store_distribute (year_month,store_code, update_date) values ({{row.play_time}}, {{row.concert_id}}, TO_TIMESTAMP('{{row.timestamp_type}}', 'YYYY-MM-DDTHH:MI:SS.US') ); {% endfor %}\""
+            }
         )
     }
 )
 public class Query extends AbstractJdbcQuery implements RunnableTask<AbstractJdbcQuery.Output>, AutoCommitInterface {
     protected final Boolean autoCommit = true;
 
+    @Schema(
+        title = "Add sqlite file.",
+        description = "The file must be from Kestra's internal storage"
+    )
+    @PluginProperty(dynamic = true)
+    protected String sqliteFile;
+
+    @Getter(AccessLevel.NONE)
+    protected transient Path workingDirectory;
+
     @Override
     public Properties connectionProperties(RunContext runContext) throws Exception {
         Properties properties = super.connectionProperties(runContext);
 
+        URI url = URI.create((String) properties.get("jdbc.url"));
+
+        // get file name from url scheme parts
+        String filename = url.getSchemeSpecificPart().split(":")[1];
+
+        Path path = workingDirectory.toAbsolutePath().resolve(filename);
```

Comment:
There is a `runContext.resolve(path)` method that check for path traversal, please use it.